akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/EventsourcedRequestingRecoveryPermit.scala

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright (C) 2016-2018 Lightbend Inc. <http://www.lightbend.com/>
+ */
+package akka.persistence.typed.internal
+
+import akka.actor.typed.Behavior
+import akka.actor.typed.scaladsl.Behaviors.MutableBehavior
+import akka.actor.typed.scaladsl.{ ActorContext, StashBuffer }
+import akka.event.Logging
+import akka.persistence._
+import akka.persistence.typed.internal.EventsourcedBehavior.WriterIdentity
+import akka.{ actor ⇒ a }
+
+abstract class EventsourcedRequestingRecoveryPermit[Command, Event, State](val context: ActorContext[Any])
+  extends MutableBehavior[Any]
+  with EventsourcedBehavior[Command, Event, State]
+  with EventsourcedStashManagement {
+
+  import Behavior.same
+  import akka.actor.typed.scaladsl.adapter._
+
+  // has to be lazy, since we want to obtain the persistenceId
+  protected lazy val log = Logging(context.system.toUntyped, this)
+
+  override protected val internalStash = StashBuffer[Any](32) // FIXME what size?
```

Comment:
conceptually strange to create the stash in RequestingRecoveryPermit?

should be configurable, but I think we have to use a rather high default